sdk/core/core-amqp/src/util/typeGuards.ts

```diff
@@ -45,9 +45,5 @@ export function objectHasProperty<Thing extends unknown, PropertyName extends st
   thing: Thing,
   property: PropertyName
 ): thing is Thing & Record<PropertyName, unknown> {
-  if (!(property in thing)) {
-    return false;
-  }
-
-  return true;
+  return typeof thing === "object" && property in (thing as Object);
```

Comment:
cc @chradek

Reply:
Thanks, I happened to run into this on Friday because vscode was using 4.2. I think I had to cast `thing` as `any` to keep eslint happy though.

I also wish TypeScript could see that `thing` is both defined (and not null), and of type `object` based on the if statement above :(